tests/platform_tests/fwutil/conftest.py

```diff
@@ -0,0 +1,107 @@
+import tarfile
+import json
+import pytest
+import logging
+
+from random import randrange
+
+from fwutil_common import show_firmware
+
+logger = logging.getLogger(__name__)
+
+DUT_HOME="/home/admin"
+DEVICES_PATH="/usr/share/sonic/device"
+FS_PATH_TEMPLATE = "/host/image-{}/fs.squashfs"
+FS_RW_TEMPLATE = "/host/image-{}/rw"
+FS_WORK_TEMPLATE = "/host/image-{}/work"
+FS_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-fs"
+OVERLAY_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-overlay"
+
+def check_path_exists(path):
+    return duthost.stat(path = path)["stat"]["exists"] 
+
+def pytest_generate_tests(metafunc):
+    val = metafunc.config.getoption('--fw-pkg')
+    if 'fw_pkg_name' in metafunc.fixturenames and val is not None:
+        metafunc.parametrize('fw_pkg_name', [val], scope="module")
+
+@pytest.fixture(scope='module')
+def fw_pkg(fw_pkg_name):
+    logger.info("Unpacking firmware package to ./firmware")
+    os.mkdir("firmware")
+    with tarfile.open(fw_pkg_name, "r:gz") as f:
+        f.extractall("./firmware/")
+        with open('./firmware/firmware.json', 'r') as fw:
+            fw_data = json.load(fw)
+            yield fw_data
+    subprocess.call("rm -rf firmware", shell=True)
+
+@pytest.fixture(scope='function')
+def random_component(duthost, fw_pkg):
+    chass = show_firmware(duthost)["chassis"].keys()[0]
+    components = fw_pkg["chassis"].get(chass, {}).get("component", []).keys()
+
+    if len(components) == 0:
+        pytest.skip("No suitable components found in config file for platform {}.".format(duthost.facts['platform']))
+
+    return components[randrange(len(components))] 
+
+@pytest.fixture(scope='function')
+def host_firmware(localhost, duthost):
+    logger.info("Starting local python server to test URL firmware update....")
+    comm = "python3 -m http.server --directory {}".format(os.path.join(DEVICES_PATH, 
+        duthost.facts['platform']))
+    task, res = duthost.command(comm, module_ignore_errors=True, module_async=True)
+    yield "http://localhost:8000/"
+    task.terminate()
```

Comment:
@alexrallen please double check if it really stops the server. I remember that there was an issue with async mode.

Reply:
It does not. Fixed.